client/html/src/Client/Html/Catalog/Product/Standard.php

```diff
@@ -0,0 +1,468 @@
+		 *
+		 * @param array List of domain names
+		 * @since 2019.06
+		 * @category Developer
+		 * @see client/html/catalog/domains
+		 * @see client/html/catalog/detail/domains
+		 * @see client/html/catalog/stage/domains
+		 * @see client/html/catalog/lists/domains
+		 */
+		$domains = $config->get( 'client/html/catalog/product/domains', $domains );
+
+		/** client/html/catalog/product/product-codes
+		 * List of codes of products to load for the current list.
+		 * Should be set dynamically through some integration plugin,
+		 * to allow a list of products with configurable products.
+		 *
+		 * @param string List of codes of products to load for the current list
+		 * @since 2019.06
+		 * @category Developer
+		 */
+		$productCodes = $config->get( 'client/html/catalog/product/product-codes', [] );
+
+		$products = \Aimeos\Controller\Frontend::create( $context, 'product' )
+			->compare( '==', 'product.code', $productCodes )
+			->slice( 0, count( $productCodes ) )
+			->uses( $domains )
+			->search();
+
+		// Sort products by the order given in the configuration "client/html/catalog/product/product-codes".
+		$productCodesOrder = array_flip( $productCodes );
+		usort( $products, function ( $a, $b ) use ( $productCodesOrder ) {
+			return $productCodesOrder[$a->getCode()] - $productCodesOrder[$b->getCode()];
+		} );
+
+
+		if ( $config->get( 'client/html/catalog/product/basket-add', false ) ) {
+			foreach ($products as $product) {
+				if ( $product->getType() === 'select' ) {
+					$productItems += $product->getRefItems( 'product', 'default', 'default' );
+				}
+			}
+		}
+
+		/** client/html/catalog/product/stock/enable
+		 * Enables or disables displaying product stock levels in product list views
+		 *
+		 * This configuration option allows shop owners to display product
+		 * stock levels for each product in list views or to disable
+		 * fetching product stock information.
+		 *
+		 * The stock information is fetched via AJAX and inserted via Javascript.
+		 * This allows to cache product items by leaving out such highly
+		 * dynamic content like stock levels which changes with each order.
+		 *
+		 * @param boolean Value of "1" to display stock levels, "0" to disable displaying them
+		 * @since 2019.06
+		 * @category User
+		 * @category Developer
+		 * @see client/html/catalog/detail/stock/enable
+		 * @see client/html/catalog/stock/url/target
+		 * @see client/html/catalog/stock/url/controller
+		 * @see client/html/catalog/stock/url/action
+		 * @see client/html/catalog/stock/url/config
+		 */
+
+		if ( !empty( $products ) && (bool)$config->get( 'client/html/catalog/product/stock/enable', true ) === true ) {
+			$view->itemsStockUrl = $this->getStockUrl( $view, $products + $productItems );
+		}
+
+		// Delete cache when products are added or deleted even when in "tag-all" mode
+		$this->addMetaItems( $products + $productItems, $expire, $tags, ['product'] );
+
+		$view->listProductItems = $products;
```

Comment:
You should only pass $products as `$view->productItems`

Reply:
See below. Changed to `productProductItems`.